app/api/auth.py

```diff
@@ -19,54 +19,135 @@
 
 @bp.route('/register', methods=['POST'])
 def register():
-    if not request.json:
-        abort(500)
-    name = request.json.get("name", None)
-    username = request.json.get("username", None)
-    password = request.json.get("password", None)
-    if not name or not username or not password:
-        return jsonify({"msg": "Invalid Request"}), 400
-
-    user = mongo.db.users.count({
-        "username": username
-    })
-    if user > 0:
-        return jsonify({"msg": "Username already taken"}), 500
-
-    id = mongo.db.users.insert_one({
-        "name": name,
-        "password": pbkdf2_sha256.hash(password),
-        "username": username
-    }).inserted_id
-    return jsonify(str(id))
+   if not request.json:
+       abort(500)
+   name = request.json.get("name", None)

```

Comment:
there is no condition here. e.g registration route should not work if the setting is on? 
same for login